llvm-external-projects/iree-dialects/lib/Dialect/VectorExt/IR/VectorExtOps.cpp

```diff
@@ -36,6 +36,22 @@ LogicalResult LayoutConflictResolutionOp::verify() {
   return failure();
 }
 
+// to_simd -> to_simt
+OpFoldResult ToSIMDOp::fold(FoldAdaptor) {
+  if (auto simtOp = getOperand().getDefiningOp<ToSIMTOp>()) {
```

Comment:
Hmm, not sure if this is 100% safe, may need dyn_cast_or_null instead. (i.e Very unlikely case but plausible, if the argument was coming in from a blockArg it might not be able to summon getDefiningOp)

Reply:
getDefiningOp returns nullptr if the arguement is not the operation it wants or if it's a block argument. so it should be safe.

Reply:
yeah definingOp is fine but not so sure about the <> cast after. IIRC we've had some segfault issues with that(and or regular dyn_cast). At the end we need to replace them with dyn_cast_or_null before we

Reply:
Right. getDefiningOp used dyn_cast_or_null internally so it's the same thing.

Reply:
Ah I see, maybe it was issue with dyn_cast only then, not templates getDefiningOp.